Dockerfile

```diff
@@ -1,21 +1,27 @@
-FROM python:3.7-alpine
+FROM python:3.7-slim-stretch
```

Comment:
We can utilize multi-stage builds here to end up with a "cleaner" final image as shown below. The same is probably possible for the CI image, but I haven't looked at that yet.

```
# First build the Helm binding
FROM golang:1.12.9-stretch AS go-builder

RUN mkdir /kapitan
WORKDIR /kapitan
COPY . .

RUN ./kapitan/inputs/helm/build.sh

# Build the virtualenv for Kapitan
FROM python:3.7-slim-stretch AS python-builder

COPY --from=go-builder /kapitan /kapitan
WORKDIR /kapitan

ENV PATH="/opt/venv/bin:${PATH}"

RUN apt-get update \
  && apt-get install --no-install-recommends -y build-essential gcc \
  && python -m venv /opt/venv \
  && pip install --upgrade pip setuptools wheel \
  && pip install -r requirements.txt \
  && ./kapitan/inputs/helm/build.sh \
  && pip install .

# Final image with virtualenv built in previous step
FROM python:3.7-slim-stretch

COPY --from=python-builder /opt/venv /opt/venv

ENV PATH="/opt/venv/bin:${PATH}"
ENV SEARCHPATH="/src"
VOLUME ${SEARCHPATH}
WORKDIR ${SEARCHPATH}

# Install runtime dependencies and run as a non-root user for good measure
RUN apt-get update \
  && apt-get install --no-install-recommends -y git \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/* \
  && useradd --create-home --no-log-init --user-group kapitan

USER 1000

ENTRYPOINT ["kapitan"]
```

The build script just needs to be modified as shown below in order to allow it to work for the multi-stage builds:

```sh
#!/usr/bin/env bash

cd $(dirname "$0")
pwd
so_name=libtemplate.so

# Compile the binding if a Go runtime exists
if [[ -z $(which go) ]]; then
    echo "[WARN] go is not available on this system -- skipping Helm binding build!"
else
    go build -buildmode=c-shared -o $so_name template.go
fi

# Validate that the compiled binding exists
if [[ -e $so_name ]]; then
    echo "$so_name built successfully"
else
    echo "[ERROR] $so_name does not exist!"
    exit 1
fi

# Compile the Python ffi binding if Python is available
if [[ -z $(which python3) ]]; then
    echo "[WARN] python3 is not available on this system -- skipping cffi build!"
else
    echo "Building the Python binding using cffi"
    python3 cffi_build.py
fi

# We made it! Time to exit cleanly...
exit 0
```

Reply:
The multi-stage build would be awesome and we should definitely add it, but ideally in a separate PR since it's semantically separate from the current one. :) 
Feel free to open a PR after this is merged or I'll have a look at it after these are fixed and a new RC is out. :+1:

Reply:
Makes sense! I'll open the PR when possible.